src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/domain/CreatedAllocationDetails.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsworkload.domain
+
+import uk.gov.justice.digital.hmpps.hmppsworkload.client.dto.AllocationDetail
+import uk.gov.justice.digital.hmpps.hmppsworkload.client.dto.Name
+import uk.gov.justice.digital.hmpps.hmppsworkload.client.dto.StaffMember
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.CaseDetailsEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.EventManagerEntity
+import java.time.ZonedDateTime
+
+data class CreatedAllocationDetails(val cases: List<CreatedAllocationDetail>) {
+  companion object {
+    fun from(eventManagers: List<EventManagerEntity>, eventManagerDetails: Map<String, AllocationDetail>, caseDetails: Map<String, CaseDetailsEntity>): CreatedAllocationDetails {
+      return CreatedAllocationDetails(
+        eventManagers
+          .filter { eventManagerDetails.containsKey(it.crn) && caseDetails.containsKey(it.crn) }
```

Comment:
Probably not that important but this code means that if anything is missing from a record then they won't see the allocation at all.

Now, I have no idea how possible it would be to not find matching details for a CRN (maybe they have left probation? maybe there is an inconsistency in Delius?) but it might be worth putting something in the logs so at least we know something might be wrong.

Reply:
yeah, its an extraordinary unlikely scenario but I did think if it happened and we didn't have the filter the error would take ages to debug and resolve